### PR TITLE
Paginator re-implementation

### PR DESCRIPTION
**Pull request by nseckinoral**
-Current frame will not move until you go to a page from another frame
on purpose. Eg. On any page in 1-5 frame you'll always see the same
frame. If you go to 6th page, you'll see 6-10 frame and it goes on.

-"<" and ">" buttons work as "previous page" and "next page"
respectively

-"<<" and ">>" buttons work as "previous frame" and "next frame"
respectively. Eg. If you're on 1-5 frame and hit ">>" button, you'll see
6-11 frame with the same index.

-If the last index can't be restored, user will be redirected to the
last page of the current frame.

### CODE REVIEW

**Comment by nseckinoral** [comment]
Steps for testing:

1.  (Optional)  Go to appSettings.json file under XOMNI.CMS.FrontEnd/src and update "PaginatorFrameSize" as you like. This property sets the maximum number of pages that should be shown. Make sure it's a number and greater than zero. Default value is 5. 

2. Navigate to the html file of the page that you'd like to see the paginator component. Edit the file and copy paste the code below to wherever you want it to appear. Bottom of the page would be a better idea since it's a pager component. You can change the page count by editing the "totalPageCount" parameter.
```
 <paginator params="{totalPageCount: 6}"></paginator>
```
3.Save/Build and open the page in your browser.

NOTE: If you have any tags that's closed with a "/" at the end, just before the pager component, make sure to close it properly. It causes the component to not load properly.

Bad Example:
```
  <validation-summary params="{errors:getValidationErrors()}" />
  <paginator params="{totalPageCount: 6}" />
```
Good Example:
```
  <validation-summary params="{errors:getValidationErrors()}"></validation-summary>
  <paginator params="{totalPageCount: 6}"></paginator>
```

File: src/XOMNI.CMS.FrontEnd/src/components/paginator/paginator.ts [hunk 1]
```diff
@@ -149,35 +171,26 @@ export class viewModel extends cms.infrastructure.baseViewModel {
                 results[urlParameters[i]] = urlParameters[urlParameters[i]];
             }
         }
-        
+
         return results;
     }
 
     public calculateBoundry(frameSize: number, totalPageCount: number, currentPage: number) {
-
```
Comment (ilkayilknur) [inline src/XOMNI.CMS.FrontEnd/src/components/paginator/paginator.ts]: We need to implement this method more efficiently. Details are discussed in an offline meeting.

**Comment by nseckinoral** [comment]
@ilkayilknur I've refactored the method to be more efficient as we discussed. Could you please take another look?

**Comment by nseckinoral** [comment]
@ilkayilknur Thanks for the heads up! Fixed both.

**Pull request merged**